dashboard/src/scenes/stats/charts.js

```diff
@@ -77,11 +77,26 @@ export const CustomResponsivePie = ({ data = [], title, onItemClick, help }) =>
 
 const getItemValue = (item) => Object.values(item)[1];
 
-export const CustomResponsiveBar = ({ title, data, categories, onItemClick, axisTitleX, axisTitleY, help }) => {
-  const total = data.reduce((sum, item) => sum + getItemValue(item), 0);
+export const CustomResponsiveBar = ({ title, data, categories, onItemClick, axisTitleX, axisTitleY, isMultiChoice, originalDatasetLength, help }) => {
+  // if we have too many categories with small data, we see nothing in the chart
+  // so we filter this way:
+  // - keep the first 15 categories whatever
+  // - keep the others only if they represent more than 1% of the total
+  const chartData = data
+    .filter((c) => c.name !== 'Non renseigné')
+    .filter((item, index) => {
+      if (index < 15) return true;
+      return item[item.name] / originalDatasetLength > 0.01;
+    });
```

Comment:
Du coup ce serait une sorte de sort ici genre:

```suggestion
const chartData = data
  .filter((c) => c.name !== 'Non renseigné')
  .sort((a, b) => b[b.name] - a[a.name])
  .slice(0, 15);
```